packages/web-api/src/methods.ts

```diff
@@ -1481,8 +1481,10 @@ export interface ChatScheduledMessagesListArguments extends WebAPICallOptions, T
 }
 cursorPaginationEnabledMethods.add('chat.scheduledMessages.list');
 export interface ChatUnfurlArguments extends WebAPICallOptions, TokenOverridable {
```

Comment:
@stevengill @srajiang and rest of the team: what about using a union type in order to represent the situation in this API of "either provide `source` and `unfurl_id`, or `channel` and `ts`"? It's a bit more complicated but it should make it clear to developers that one needs to provide either pair of arguments, but can't mix and match between the four properties (and also disambiguates a bit since we remove the superfluous optionality of all four props).

I.e. something like:

```
interface ChannelAndTS {
  channel: string;
  ts: string;
}

interface SourceAndUnfurlID {
  source: 'composer' | 'conversations_history';
  unfurl_id: string;
}

type UnfurlArguments = (ChannelAndTS | SourceAndUnfurlID) & {
  unfurls: string[];
  whatever?: string;
};

const _channelArgs: UnfurlArguments = {
  channel: 'C12345',
  ts: '1234.56',
  unfurls: ['yes', 'no'],
};

const _sourceArgs: UnfurlArguments = {
  source: 'composer',
  unfurl_id: '12345',
  unfurls: ['maybe', 'so'],
};
```

Reply:
Yeah, you can definitely do that - we did something like that for next-gen views.open method with `interactivity_pointer` vs. 'trigger_id'.

Reply:
This is definitely better for TS users